File: scripts/kanxue_checkin.js
const axios = require('axios');
const PUSHPLUS = process.env.PUSHPLUS;
const KANXUE_COOKIE = process.env.KANXUE_COOKIE;

// 添加日志输出工具类
class Logger {
    static info(msg) {
        console.log(`ℹ️ ${new Date().toLocaleString('zh-CN')} [INFO] ${msg}`);
    }

    static success(msg) {
        console.log(`✅ ${new Date().toLocaleString('zh-CN')} [SUCCESS] ${msg}`);
    }

    static warning(msg) {
        console.log(`⚠️ ${new Date().toLocaleString('zh-CN')} [WARNING] ${msg}`);
    }

    static error(msg) {
        console.log(`❌ ${new Date().toLocaleString('zh-CN')} [ERROR] ${msg}`);
    }
}

const checkCOOKIE = async () => {
    Logger.info("检查Cookie配置...");

    if (!KANXUE_COOKIE || !KANXUE_COOKIE.length) {
        Logger.error('不存在 KANXUE_COOKIE，请重新检查');
        return false;
    }

    const pairs = KANXUE_COOKIE.split(/\s*;\s*/);
    for (const pairStr of pairs) {
        if (!pairStr.includes('=')) {
            Logger.error(`存在不正确的 COOKIE，请重新检查`);
            return false;
        }
    }

    Logger.success("Cookie检查通过");
    return true;
}

const checkIn = async () => {
    Logger.info("=" * 60);
    Logger.info("🚀 看雪论坛自动签到开始");
    Logger.info("=" * 60);

    const options = {
        method: 'post',
        url: `https://bbs.kanxue.com/user-signin.htm`,
        headers: {
            'User-Agent': 'HD1910(Android/7.1.2) (pediy.UNICFBC0DD/1.0.5) Weex/0.26.0 720x1280',
            'Cookie': KANXUE_COOKIE,
            'Connection': 'keep-alive',
            'Accept': '*/*'
        }
    };

    Logger.info(`📡 目标地址: ${options.url}`);
    Logger.info(`📦 Cookie长度: ${KANXUE_COOKIE.length} 字符`);

    return axios(options).catch(error => {
        if (error.response) {
            Logger.error(`请求失败，状态码: ${error.response.status}`);
            Logger.info(`响应数据: ${JSON.stringify(error.response.data)}`);
        } else if (error.request) {
            Logger.error("网络请求失败，请检查网络连接");
            Logger.info(`请求详情: ${error.request}`);
        } else {
            Logger.error('发送请求时出错');
            Logger.info(`错误信息: ${error.message}`);
        }
        Logger.info(`错误配置: ${JSON.stringify(error.config)}`);
    });
};

const sendMsg = async (msg, code) => {
    const token = PUSHPLUS;
    if (!token) {
        Logger.warning("未配置 PUSHPLUS，跳过通知推送");
        return;
    }

    let title = '看雪论坛签到';
    let content;

    if (code == 0){
        content = `🎉🎉🎉签到成功，获得${msg}雪币`;
    } else {
        content = `📢您已签到，无需重复签到`;
    }

    Logger.info("准备推送签到结果通知...");

    const data = {
        token,
        title: title,
        content: content,
        template: 'json'
    };

    Logger.info(`推送数据: ${JSON.stringify({
        ...data,
        token: data.token.replace(/^(.{1,4})(.*)(.{4,})$/, (_, a, b, c) => a + b.replace(/./g, '*') + c)
    })}`);

    return axios({
        method: 'post',
        url: `http://www.pushplus.plus/send`,
        data
    }).then(response => {
        if (response.data && response.data.code === 200) {
            Logger.success("通知推送成功");
        } else {
            Logger.warning(`推送服务返回异常: ${JSON.stringify(response.data)}`);
        }
        return response;
    }).catch((error) => {
        if (error.response) {
            Logger.warning(`PUSHPLUS推送 请求失败，状态码：${error.response.status}`);
        } else if (error.request) {
            Logger.warning('PUSHPLUS推送 网络错误');
        } else {
            Logger.error(`通知推送异常: ${error.message}`);
        }
    });
};

const start = async () => {
    try {
        console.log(`
    ╔═══════════════════════════════════════╗
    ║     看雪论坛自动签到 - GitHub版       ║
    ╚═══════════════════════════════════════╝
        `);

        Logger.info("开始执行看雪论坛自动签到任务...");

        const checkCOOKIE_result = await checkCOOKIE();
        if (!checkCOOKIE_result) {
            Logger.error("Cookie检查失败，终止签到流程");
            return;
        }

        Logger.info("Cookie验证通过，开始签到请求...");
        const checkIn_result = await checkIn();

        if (!checkIn_result || !checkIn_result.data) {
            Logger.error("签到请求无响应，请检查网络或Cookie是否失效");
            return;
        }

        const message = checkIn_result.data.message;
        const code = checkIn_result.data.code;

        Logger.info(`签到响应: 状态码=${code}, 消息=${message}`);

        if (code == 0 || message == '您今日已签到成功') {
            if (code == 0) {
                Logger.success('签到成功');
                const rewardMatch = message.match(/获得(\d+)雪币/);
                if (rewardMatch) {
                    Logger.info(`🎉 签到成功，获得 ${rewardMatch[1]} 雪币`);
                }
            } else {
                Logger.info(message);
            }

            if (!PUSHPLUS || !PUSHPLUS.length) {
                Logger.warning('未配置 PUSHPLUS，跳过结果通知');
            } else {
                await sendMsg(message, code);
            }

            Logger.success("签到任务执行完成");
        } else {
            Logger.error(`签到失败: ${message}`);
        }
    } catch (error) {
        Logger.error(`签到过程中发生异常: ${error.message}`);
        Logger.info(error.stack);
    } finally {
        Logger.info("=" * 60);
    }
}

start();

File: scripts/kanxue_checkin.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
看雪论坛自动签到脚本 - GitHub Actions版本
"""

import requests
import json
import os
import sys
from datetime import datetime
import urllib3

# 禁用SSL警告
urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)

# 状态文件路径
STATUS_FILE = "status/status_kanxue.json"

def load_today_status():
    """加载今日签到状态"""
    if not os.path.exists(STATUS_FILE):
        return False

    try:
        with open(STATUS_FILE, 'r', encoding='utf-8') as f:
            data = f.read().strip()
            if not data:
                return False
            status = json.loads(data)
            # 检查是否是今天的记录
            today = datetime.now().strftime('%Y-%m-%d')
            if status.get('date') == today and status.get('success'):
                print(f"✅ 今日({today})已成功签到，跳过本次运行")
                return True
    except Exception as e:
        print(f"⚠️ 读取状态文件失败: {e}")

    return False

def save_today_status(success, message=""):
    """保存今日签到状态"""
    today = datetime.now().strftime('%Y-%m-%d')
    status = {
        'date': today,
        'success': success,
        'message': message,
        'timestamp': datetime.now().strftime('%Y-%m-%d %H:%M:%S')
    }

    # 确保status目录存在
    os.makedirs(os.path.dirname(STATUS_FILE), exist_ok=True)
    
    try:
        with open(STATUS_FILE, 'w', encoding='utf-8') as f:
            f.write(json.dumps(status, ensure_ascii=False, indent=2))
        print(f"💾 状态已保存: {status}")
    except Exception as e:
        print(f"⚠️ 保存状态失败: {e}")

def check_cookie(cookie):
    """检查Cookie格式"""
    if not cookie or not len(cookie):
        error_msg = "❌ 错误: Cookie为空"
        print(error_msg)
        return False, error_msg

    pairs = cookie.split(';')
    for pair_str in pairs:
        pair = pair_str.strip()
        if pair and '=' not in pair:
            error_msg = f"❌ Cookie格式错误: {pair}"
            print(error_msg)
            return False, error_msg

    print("✅ Cookie检查通过")
    return True, "Cookie格式正确"

def kanxue_checkin():
    """看雪论坛签到主函数"""
    
    # 从环境变量读取配置
    cookie = os.environ.get('KANXUE_COOKIE')
    pushplus_token = os.environ.get('PUSHPLUS_TOKEN')

    if not cookie:
        error_msg = "❌ 错误: 未找到KANXUE_COOKIE环境变量"
        print(error_msg)
        save_today_status(False, error_msg)
        return False

    # 检查Cookie格式
    cookie_valid, cookie_msg = check_cookie(cookie)
    if not cookie_valid:
        save_today_status(False, cookie_msg)
        return False

    url = "https://bbs.kanxue.com/user-signin.htm"
    
    headers = {
        'User-Agent': 'HD1910(Android/7.1.2) (pediy.UNICFBC0DD/1.0.5) Weex/0.26.0 720x1280',
        'Cookie': cookie,
        'Connection': 'keep-alive',
        'Accept': '*/*'
    }

    print("=" * 60)
    print(f"🚀 看雪论坛自动签到 - {datetime.now().strftime('%Y-%m-%d %H:%M:%S')}")
    print("=" * 60)
    print(f"📡 目标地址: {url}")
    print(f"📦 Cookie长度: {len(cookie)} 字符")

    try:
        response = requests.post(
            url,
            headers=headers,
            timeout=30,
            verify=False
        )

        print(f"📊 HTTP状态码: {response.status_code}")

        if response.status_code == 200:
            try:
                data = response.json()
                print(f"📄 响应数据: {json.dumps(data, ensure_ascii=False, indent=2)}")

                message = data.get('message', '')
                code = data.get('code')

                if code == 0 or message == '您今日已签到成功':
                    if code == 0:
                        success_msg = f"签到成功！获得{message}雪币"
                        print(f"🎉 {success_msg}")
                    else:
                        success_msg = "今日已签到，无需重复签到"
                        print(f"ℹ️ {success_msg}")

                    # 发送推送通知
                    if pushplus_token:
                        send_pushplus(pushplus_token, success_msg, code)
                    else:
                        print("⚠️ 未设置PUSHPLUS_TOKEN，跳过推送")

                    save_today_status(True, success_msg)
                    return True
                else:
                    error_msg = f"签到失败: {message}"
                    print(f"❌ {error_msg}")
                    save_today_status(False, error_msg)
                    return False

            except json.JSONDecodeError as e:
                error_msg = f"JSON解析失败: {response.text[:200]}"
                print(f"❌ {error_msg}")
                save_today_status(False, error_msg)
                return False
        else:
            error_msg = f"HTTP {response.status_code}: {response.text[:200]}"
            print(f"❌ {error_msg}")
            save_today_status(False, error_msg)
            return False

    except requests.exceptions.Timeout:
        error_msg = "请求超时"
        print(f"❌ {error_msg}")
        save_today_status(False, error_msg)
        return False

    except requests.exceptions.ConnectionError as e:
        error_msg = f"网络连接失败: {str(e)}"
        print(f"❌ {error_msg}")
        save_today_status(False, error_msg)
        return False

    except Exception as e:
        error_msg = f"未知错误: {type(e).__name__} - {str(e)}"
        print(f"❌ {error_msg}")
        import traceback
        traceback.print_exc()
        save_today_status(False, error_msg)
        return False

def send_pushplus(token, msg, code):
    """发送PushPlus通知"""
    if code == 0:
        title = '看雪论坛签到成功'
        content = f'🎉🎉🎉签到成功，获得{msg}雪币'
    else:
        title = '看雪论坛签到提醒'
        content = '📢您已签到，无需重复签到'

    data = {
        "token": token，
        "title": title，
        "content": content,
        "template": "json"
    }

    # 打印脱敏的推送数据（用于调试）
    debug_data = data.copy()
    if debug_data.get('token'):
        token_str = debug_data['token']
        if len(token_str) > 8:
            debug_data['token'] = token_str[:4] + '*' * (len(token_str) - 8) + token_str[-4:]
    print(f'📤 推送数据: {debug_data}')

    try:
        response = requests.post(
            'http://www.pushplus.plus/send'，
            json=data,
            timeout=10
        )
        
        if response。status_code == 200:
            result = response.json()
            print(f'✅ PushPlus推送成功: {result.get("msg", "未知")}')
        else:
            print(f'⚠️ PushPlus推送失败，状态码: {response.status_code}')
    except requests。exceptions。RequestException as e:
        print(f'⚠️ PushPlus推送异常: {str(e)}')
    except Exception as e:
        print(f'⚠️ PushPlus推送未知错误: {str(e)}')

def main():
    """主函数"""
    # 检查今日是否已成功签到
    if load_today_status():
        print("✅ 今日已完成签到，无需重复运行")
        sys.exit(0)

    # 执行签到
    success = kanxue_checkin()

    print("=" * 60)
    if success:
        print("✅ 签到任务完成")
        sys.exit(0)
    else:
        print("❌ 签到任务失败")
        sys.exit(1)

if __name__ == "__main__":
    main()
